src/cmd/ksh93/sh/path.c

```diff
@@ -1178,20 +1177,14 @@ static void exscript(Shell_t *shp, char *path, char *argv[], char *const *envp)
             n = 10;
         }
     }
-    savet = *--argv;
     *argv = path;
-    if (err == EACCES && sh_access(e_suidexec, X_OK) < 0) {
-        errno = EACCES;
-        return;
-    }
-    path_pfexecve(shp, e_suidexec, argv, envp, 0);
+    if (err == EACCES) return;
```

Comment:
How does this function handle executing a script if `/etc/suid_exec` is not present ? I would appreciate a comment in the source code here.

Reply:
I'm going to revert this part of the change. Not because it's wrong. As far as I can tell via code inspection and testing this part of the change is itself okay. But looking more closely at the surrounding code I think the original code is broken. Take a look at the block inside

```
if ((euserid = geteuid()) != shp->gd->userid) {
```

It creates a temp file and assigns fd 10 to it. It then falls thru and calls `path_pfexecve()` to execute */etc/suid_exec* without any reference to that temp file. If the execve fails it then falls through to the `fail:` label and the temp file created earlier isn't used.

So instead I'm going to leave a "TODO" comment to remind someone to rewrite this function.